op-challenger/fault/factory_test.go

```diff
@@ -0,0 +1,189 @@
+package fault
+
+import (
+	"context"
+	"errors"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	gameCountErr = errors.New("game count error")
+	gameIndexErr = errors.New("game index error")
+)
+
+// TestGameLoader_FetchAllGames tests that the game loader correctly fetches all games.
+func TestGameLoader_FetchAllGames(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name        string
+		caller      *mockMinimalDisputeGameFactoryCaller
+		expectedErr error
+		expectedLen int
+	}{
+		{
+			name:        "success",
+			caller:      newMockMinimalDisputeGameFactoryCaller(10, false, false),
+			expectedErr: nil,
+			expectedLen: 10,
+		},
+		{
+			name:        "game count error",
+			caller:      newMockMinimalDisputeGameFactoryCaller(10, true, false),
+			expectedErr: gameCountErr,
+			expectedLen: 0,
+		},
+		{
+			name:        "game index error",
+			caller:      newMockMinimalDisputeGameFactoryCaller(10, false, true),
+			expectedErr: gameIndexErr,
+			expectedLen: 0,
+		},
+		{
+			name:        "no games",
+			caller:      newMockMinimalDisputeGameFactoryCaller(0, false, false),
+			expectedErr: nil,
+			expectedLen: 0,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+
+			loader := NewGameLoader(test.caller)
+
+			games, err := loader.FetchAllGames(context.Background())
+			require.Equal(t, test.expectedErr, err)
```

Comment:
`require.ErrorIs` would work better here so it can handle the error being wrapped with extra context.